src/SymbolFinder.php

```diff
@@ -72,13 +97,19 @@ public function enterNode(Node $node)
                 new Position($node->getAttribute('endLine') - 1, $node->getAttribute('endColumn'))
             )
         );
-        $symbol->containerName = $this->containerName;
-        $this->containerName = $symbol->name;
+        $symbol->containerName = $containerName;
         $this->symbols[] = $symbol;
     }
 
     public function leaveNode(Node $node)
     {
-        $this->containerName = null;
+        array_pop($this->nodeStack);
+        array_pop($this->nameStack);
+
+        // if we leave a method or function, decrease the function counter
+        $class = get_class($node);
+        if ($class === Node\Stmt\Function_::class || $class === Node\Stmt\ClassMethod::class) {
```

Comment:
No particular reason, just reused the $class variable. Instanceof would probably make the purpose clearer.